### PR TITLE
[SYCL] Added E2E test for kernel_bundle::get_kernel<KernelName>()

### PR DESCRIPTION
**Pull request by andylshort**
This patch adds the test to ensure the functionality of the `template <typename KernelName> kernel get_kernel() const` function that was added in patch https://github.com/intel/llvm/pull/8294

An E2E test was required to ensure the correct information was present to compare and test the kernels against one another via `get_info`, as `operator==` isn't guaranteed to work in this instance.

### CODE REVIEW

**Comment by andylshort** [comment]
The implementation of this has now been merged as part of https://github.com/intel/llvm/pull/8294

File: SYCL/KernelAndProgram/kernel-bundle-get-kernel.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      ExpectedKernel.get_info<sycl::info::kernel::num_args>();
+
+  auto FoundKernelContext = FoundKernel.get_info<sycl::info::kernel::context>();
+  auto ExpectedKernelContext =
+      ExpectedKernel.get_info<sycl::info::kernel::context>();
+
+  assert(FoundKernelName == ExpectedKernelName);
+  assert(FoundKernelNumArgs == ExpectedKernelNumArgs);
+  assert(FoundKernelContext == ExpectedKernelContext);
+  assert(FoundKernel.get_kernel_bundle() == ExpectedKernel.get_kernel_bundle());
```
Comment (steffenlarsen) [inline SYCL/KernelAndProgram/kernel-bundle-get-kernel.cpp]: Does the spec guarantee this to be the case? I couldn't find anything when I did a quick skim, but maybe I missed it.

Comment (andylshort) [reply SYCL/KernelAndProgram/kernel-bundle-get-kernel.cpp]: I'm not sure that it does, it was the only way that I could think to compare them for equality considering operator== isn't guaranteed to work here. I'm not sure how else to do this?

Comment (steffenlarsen) [reply SYCL/KernelAndProgram/kernel-bundle-get-kernel.cpp]: Good question. I suppose you could check that they have the same kernels, but I don't think that's even a guarantee either. I think we can live without this last check, relying on the other checks.

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Pull request merged**